src/reporting.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::Result;
+use reqwest::Client;
+use tracing::debug;
+
+use crate::models::{
+    DiscordEmbed, DiscordEmbedAuthor, DiscordWebhookEvent, MemberCreator, WebhookActionDisplay,
+    WebhookEvent,
+};
+
+pub struct DiscordReporter {
+    endpoint: String,
+    client: Client,
+}
+
+impl DiscordReporter {
+    pub fn new(endpoint: String) -> Self {
+        Self {
+            endpoint,
+            client: Client::new(),
+        }
+    }
+
+    pub async fn report(&self, event: WebhookEvent) -> Result<()> {
+        if Self::should_skip(&event) {
+            debug!("Skipping event");
+            return Ok(());
+        }
+        let mut card_name: Option<String> = None;
+        let mut card_url: Option<String> = None;
+        if let Some(card) = event.action.data.card {
+            card_name = Some(card.name);
+            card_url = Some(format!("https://trello.com/c/{}", card.short_link));
+        }
+        let discord_event = DiscordWebhookEvent {
+            embeds: vec![DiscordEmbed {
+                title: Some(event.action._type),
+                _type: Some(String::from("rich")),
```

Comment:
Some constant maybe?
```suggestion
                _type: Some(TYPE_RICH),
```